frontend/src/components/page/PageSidebarComponent.vue

```diff
@@ -1,32 +1,42 @@
 <script lang="ts">
-import { RouterLink } from "vue-router";
+import { defineComponent, onMounted, ref, watch } from 'vue';
+import { useRoute } from 'vue-router';
+import api from '@/services/api';
+
+export default defineComponent({
+  name: 'PageSidebarComponent',
+  setup() {
+    const links = ref<{ id: number; title: string; category: any }[]>([]);
+    const route = useRoute();
+
+    const fetchLinks = async () => {
+      try {
+        const response = await api.get('/page/');
+        links.value = response.data.filter((page: any) => {
+          return page.category?.slug === route.params.slug;
+        });
+      } catch (error) {
+        console.error('Nepodarilo sa načítať stránky pre sidebar.', error);
+      }
+    };
+
+    onMounted(fetchLinks);
+
+    watch(
+      () => route.params.slug,
+      () => {
+        fetchLinks();
+      }
+    );
 
-export default {
-  name: "PageSidebarComponent",
-  components: {
-    RouterLink,
-  },
-  props: {
-    slug: {
-      type: String,
-      required: true,
-    },
-    year: {
-      type: String,
-      required: true,
-    },
-  },
-  data() {
     return {
-      links: [
-        // TODO: fetch links from backend
-        { id: 1, title: "Osoby" },
-        { id: 2, title: "Koláče" },
-        { id: 3, title: "Kontakt" },
```

Comment:
hmm tu stačilo iba implementovať jednu metódu ktorá to zoberie z API namiesto toho aby to bolo pevne definované v kóde, ale to nevadí pretože frontend sa nehodnotí, hlavne že to funguje